### DISCUSSION TITLE
Switch the moclojer to reitit instead of pedestal-routes

### DISCUSSION THREAD

**Discussion by matheusfrancisco** [moclojer/moclojer#225] [General]
Currently, I run into an issue which is this one https://github.com/moclojer/moclojer/pull/206 to add swagger into moclojer, but pedestal does not offer a straightforward approach, but reitit does. But reitit does not offer a straightforward multi host then this https://github.com/metosin/reitit/issues/660 showed me the path and I create this interceptor to reproduce https://github.com/matheusfrancisco/multihost-reitit-pedestal/blob/main/src/clojure/playground/server.clj#L26 

So I see some paths.

1 - We do stay a little long without swagger, we try to figure out with pedestal-route.
2 - We migrate to reitit and then implement swagger?
3 - Or we try to implement a swagger for pedestal by our own (it could take a while)

What do you think about it?

**Comment by matheusfrancisco** [comment]
@avelino @J0sueTM

**Comment by avelino** [comment]
> 2 - We migrate to reitit and then implement swagger?

@matheusfrancisco this option makes sense to me, but we need to be clear on a few points:

- what is the difference between `pedestal-route` and `reitit`?
- what do we "lose" by switching to `reitit`?
- what do we "gain" by switching to `reitit`?

**Reply by matheusfrancisco** [reply]
**What is the difference between pedestal-route and reitit?**
Those are my opinion, what I understood about pedestal and reitit

Pedestal
**Model**: Pedestal uses an interceptor chain model, where each route can have its own series of interceptors (functions) that are executed before and after the actual handling of a request. This model allows for a high degree of customization and control over the request/response lifecycle.
**Flexibility**: Due to its interceptor model, Pedestal can be very flexible, allowing developers to insert, reorder, or remove steps in the processing of requests.
**Verbosity**: Configuring routes and interceptors in Pedestal can be more verbose compared to some other routing libraries, due to the explicitness required in defining the interceptor chains.



Reitit

**Model**: Reitit focuses on simplicity and performance, offering a more straightforward routing table definition. It supports both Clojure and Clojurescript and aims to be fast and data-driven. Also is possible to build chain interceptors using reitit.
**Ease of Use:** Reitit is often praised for its ease of use and simple setup, making it a good choice for projects where simplicity and speed are a priority.
**Data-driven**: Reitit routes are data-driven, making them easy to manipulate, generate, and introspect. This can be particularly beneficial for applications that require dynamic route generation.
**Features**: Reitit includes out-of-the-box support for Swagger/OpenAPI integration, making API documentation easier.


Both are flexible with middlewares and interceptors.
The big difference is: 
**If simplicity, performance, and ease of use are your priorities, along with good support for API documentation via Swagger/OpenAPI, then Reitit could be more suitable.**


**What do we "lose" by switching to reitit?**

By default, we're going to lose the multi-host, and we will have to implement using interceptors

**What do we "gain" by switching to reitit?**

Out of the box, open api and swagger-ui.

**Reply by avelino** [reply]
The points that make sense to migrate to Reitit are:
- **data-driven:** mainly because we support extensive use _("framework")_
- **simplicity** of use
- **performance**
- **swagger** by default

**Reply by matheusfrancisco** [reply]
I already create an issue to https://github.com/moclojer/moclojer/issues/226 .
